src/main/kotlin/io/emeraldpay/dshackle/upstream/LogsOracle.kt

```diff
@@ -0,0 +1,29 @@
+package io.emeraldpay.dshackle.upstream
+import io.emeraldpay.dshackle.config.IndexConfig
+
+import io.emeraldpay.api.proto.BlockchainOuterClass
+
+class LogsOracle(
+    var config: IndexConfig.Index,
+) {
+    val db = org.drpc.logsoracle.LogsOracle(config.rpc, config.store, config.ram_limit ?: 0L)
+
+    fun estimate(request: BlockchainOuterClass.EstimateLogsCountRequest): BlockchainOuterClass.EstimateLogsCountResponse {
+        val count = db.query(
```

Comment:
Looks like we make blocking call inside rpc scheduler.
It could lead to performance degradation of all rpc calls.
We need to move it so separate scheduler / queue / threadpool

Reply:
I was under the impression that it was called from a new sсheduler up the stack. How do you do it right?

https://github.com/p2p-org/dshackle/pull/335/files#diff-57a0331a868e4909f5bb6522d56532631611a0bbfcc2cbc7024ffbd735e12874R130